api/api.py

```diff
@@ -197,18 +196,12 @@ def dispatcher(router, request, response):
             response.headers['Content-Type'] = 'application/json; charset=utf-8'
     except webapp2.HTTPException as e:
         util.send_json_http_exception(response, str(e), e.code)
-    except Exception as e: # pylint: disable=broad-except
-        if config.get_item('core', 'debug'):
-            message = traceback.format_exc()
-        else:
-            message = 'Internal Server Error'
-        util.send_json_http_exception(response, message, 500)
```

Comment:
@kofalt I added a ref to issue #451, which could be addressed as a separate change but since this except  branch was clearly added to facilitate debug mode I removed it.  I chose to let it go up the stack because without debug mode, the behavior would always be "log the error and send a message saying 'Internal Server Error'" to the client, which we can do just by letting the exception bubble up.  Also it seems to me having 500's return JSON is not a good idea because other times we may not catch an error and thus the body may or may not be JSON encoded, which could cause issues for the requester.  Status code 500 defined as "A generic error message, given when an unexpected condition was encountered and no more specific message is suitable"